### PR TITLE
Run2 legacy v1  lhapdf

### PR DESCRIPTION
**Pull request by jedori0228**
Now we have AnalyzerTools/ directory, which contains all tool classes (e.g., CFBackgroundEstimator, FakeBackgroundEstimator, and MCCorrection. Now LHAPDFHandler and PDFReweight).

LHAPDF handler and PDFReweight from LHAPDF are added.
Usage are described in 
https://github.com/jedori0228/SKFlatAnalyzer/blob/9e28f750d2443782527e8a38f0967d140ba823c2/Analyzers/src/ExampleRun.C#L48-L77
and
https://github.com/jedori0228/SKFlatAnalyzer/blob/9e28f750d2443782527e8a38f0967d140ba823c2/Analyzers/src/ExampleRun.C#L163-L183

Xsec systematic functions added (Scale and AlphaS added. I think one should write their own function for PDFErrorSet).

### CODE REVIEW

File: Analyzers/src/AnalyzerCore.C [hunk 1]
```diff
@@ -719,6 +719,32 @@ std::vector<FatJet> AnalyzerCore::SmearSDMassFatJets(std::vector<FatJet> jets, i
...
+double AnalyzerCore::GetPDFAlphaSError(){
+
+  if(IsDATA) return 0.;
+  if(PDFWeights_AlphaS->size()!=2) return 0.;
+
+  return ( PDFWeights_AlphaS->at(1)-PDFWeights_AlphaS->at(0) ) / 2.;
+
+}
+//==== Renorm/Fact Scale
+double AnalyzerCore::GetScaleError(){
```
Comment (sungbinoh) [inline Analyzers/src/AnalyzerCore.C]: This function on my previous commit was wrong.
For the scale errors, kinematics should be achieved using only one kind of scale shift and after that, envelope will be set as systematics bin by bin on histrogram, not choosing envelope event by event.
So, how about to make this function to be able to call six error with int 1~6 ?

Comment (jedori0228) [reply Analyzers/src/AnalyzerCore.C]: Hi SB, as suggested, related functions are now removed in the new commit.

Comment (sungbinoh) [reply Analyzers/src/AnalyzerCore.C]: Good! How about to add example lines on ExampleRun.C for users?

Comment (jedori0228) [reply Analyzers/src/AnalyzerCore.C]: Hi SB, this is also added!

Comment (sungbinoh) [reply Analyzers/src/AnalyzerCore.C]: Great! I will resolve conversation.

**Comment by jedori0228** [comment]
Hi John, thank you for the review!

> Comment 1:
> 
> [233f575#diff-fb7983001e1cd25f43c567e3917616ceR127](https://github.com/CMSSNU/SKFlatAnalyzer/commit/233f57523097627a5f1d9c8b791f5e42120d5781#diff-fb7983001e1cd25f43c567e3917616ceR127)
> 
> Q: Since pdfReweight is a non pointer, and variables are set in the inherited class, will the variables be set ok?
> 

I am not sure about this issue, but we can easily repalce it by a pointer. We can also change other tool objects (e.g., mcCorr, fakeEst, and cfEst from https://github.com/jedori0228/SKFlatAnalyzer/blob/0329b4d99a5a960b024368cfe57f619ebcb5efec/Analyzers/include/AnalyzerCore.h#L122-L124) 
to pointers as well. Please check the next commit which I will push soon.

> Comment 2:
> 
> [233f575#diff-07401f6d26667ca696b9ffc4e69320bbR11](https://github.com/CMSSNU/SKFlatAnalyzer/commit/233f57523097627a5f1d9c8b791f5e42120d5781#diff-07401f6d26667ca696b9ffc4e69320bbR11)
> 
> Q: Why is this line included? #INCLUDES += -I/cvmfs/cms.cern.ch/slc6_amd64_gcc630/external/lhapdf/6.2.1-fmblme/include/
> 

I found this is necessary to do "#include "LHAPDF/LHAPDF.h" in "AnalyzerTools/include/LHAPDFHandler.h"

> Comment 3:
> 
> [233f575#diff-7c5f4070e01edc39a3e8fd97057827c2R4](https://github.com/CMSSNU/SKFlatAnalyzer/commit/233f57523097627a5f1d9c8b791f5e42120d5781#diff-7c5f4070e01edc39a3e8fd97057827c2R4)
> 
> Q: Is -j4 ok in makefile? Since now Analysis depends on AnalysisTools if you compile with 4 cores will all dependancies work?
> 

I am not sure about this issue either.. Which one do you recommend?

> Comment 4:
> 
> Q Looking at make file and LHADPF I see lots of warnings from LHAPDF package, either this needs installing in SKAnalyzer directory (like with Rochester corr) or change compilation settings to stop warnings, or add flag not to compile LHADPF unless a flag is set.
> 

This is true, I also observed so many warnings, and I had to add "-Wno-ignored-qualifiers -Wno-unused-parameter" (This is done in this PR). In my understanding, sharedlibraries are already compiled, and if we use this, we don't have to compile again. But we see the warnings from LHAPDF, and does this mean we are compiling it again? Anyone know about this..?

> Comment 5:
> [233f575#diff-79b99314bbe8ca7d4c1456d05dc16354R407](https://github.com/CMSSNU/SKFlatAnalyzer/commit/233f57523097627a5f1d9c8b791f5e42120d5781#diff-79b99314bbe8ca7d4c1456d05dc16354R407)
> 
> Q: How is LHAPDF lib file created. I see the line
> ## IncludeLine += 'R__LOAD_LIBRARY({0}libLHAPDF.so)\n'.format(libdir)"
> 
> But I don't see any compilation of the lhapdf.
> Is this compiled, or copied from /cvmfs/cms.cern.ch/slc6_amd64_gcc630/external/lhapdf/6.2.1-fmblme?
> If it is copied won't it be incompatible at run time?
>

First, I missed to update setup_tmp.sh file for SNU machines. Before, we use root from "/share/apps/root_v6-12-06/bin/thisroot.sh" in SNU, but we should now use cvmfs same as in other machins. Also, "lib/libLHAPDF.so" is copied from "/cvmfs/cms.cern.ch/slc6_amd64_gcc630/cms/cmssw/CMSSW_9_4_9_cand2/external/slc6_amd64_gcc630/lib/libLHAPDF.so" and I added this in this repository.
 
> Comment 6:
> [233f575#diff-7c5f4070e01edc39a3e8fd97057827c2R8](https://github.com/CMSSNU/SKFlatAnalyzer/commit/233f57523097627a5f1d9c8b791f5e42120d5781#diff-7c5f4070e01edc39a3e8fd97057827c2R8)
> 
> INCLUDES += -I/cvmfs/cms.cern.ch/slc6_amd64_gcc630/external/lhapdf/6.2.1-fmblme/include/
> Q: Is this path correct for all servers? If on SNU/Kisti/KNU, what if someone compiles on local machine, will it compile ok on a Mac if this path does not exist?

I only considered running this analyzer in machines with cvmfs installed.

File: Analyzers/src/ExampleRun.C [hunk 2]
```diff
@@ -67,10 +67,10 @@ void ExampleRun::initializeAnalyzer(){
     LHAPDFHandler_New.AlphaSMember_Up = 102; 
     LHAPDFHandler_New.init();
 
-    pdfReweight.SetProdPDF( LHAPDFHandler_Prod.PDFCentral );
-    pdfReweight.SetNewPDF( LHAPDFHandler_New.PDFCentral );
-    pdfReweight.SetNewPDFErrorSet( LHAPDFHandler_New.PDFErrorSet );
-    pdfReweight.SetNewPDFAlphaS( LHAPDFHandler_New.PDFAlphaSDown, LHAPDFHandler_New.PDFAlphaSUp );
+    pdfReweight->SetProdPDF( LHAPDFHandler_Prod.PDFCentral );
```
[the remark this answers is not in the document]
Comment (jedori0228) [reply Analyzers/src/ExampleRun.C]: It was missing, so I added them in

https://github.com/CMSSNU/SKFlatAnalyzer/blob/ad29244c275fc011e6b6a08a3219b2404cb59231/Analyzers/src/AnalyzerCore.C#L5-L8

and

https://github.com/CMSSNU/SKFlatAnalyzer/blob/ad29244c275fc011e6b6a08a3219b2404cb59231/Analyzers/src/AnalyzerCore.C#L32-L35

**Comment by jedori0228** [comment]
I also made few updates for KISTI (minor mistakes.. just tested it works well now).

As SB and John approved this, we can now merge it.

**Pull request merged**